learning_service/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM python:3.7.10-slim
+
+RUN apt-get update \
+&& apt-get install -y --no-install-recommends git \
+&& apt-get purge -y --auto-remove \
+&& rm -rf /var/lib/apt/lists/*
+
+WORKDIR /root/
+
+ENV VIRTUAL_ENV=/root/venv
+RUN python -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
```

Comment:
I think we can skip playing around with the virtual environment inside the container and just execute `pip install -r requirements.txt` as a root. Then it's the good practice to start the process as non-root:

```dockerfile
# Make new user as root
WORKDIR /app
RUN useradd -d /app appuser && chown -R appuser: /app

# ...
# Copy files and install dependencies still as root
COPY . .
RUN pip install -r requirements.txt

# Step down from being a root before calling CMD/ENTRYPOINT
USER appuser
```